File: core/src/main/java/org/ecos/reader/core/io/files/PathImpl.java
package org.ecos.reader.core.io.files;

public class PathImpl implements Path {
    private String pathAsString;

    private PathImpl(String pathAsString) {

        this.pathAsString = pathAsString;
    }

    public static Path from(String pathAsString){
        return new PathImpl(pathAsString);
    }

    @Override
    public String asString() {
        return pathAsString;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        PathImpl path = (PathImpl) o;

        return pathAsString.equals(path.pathAsString);
    }

    @Override
    public int hashCode() {
        return pathAsString.hashCode();
    }
}

File: files/src/main/kotlin/org/ecos/core/files/PathImpl.kt
package org.ecos.core.files

class PathImpl private constructor(private val pathAsString: String) : Path {

    override fun asString(): String {
        return pathAsString
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val path = other as PathImpl?

        return pathAsString == path!!.pathAsString
    }

    override fun hashCode(): Int {
        return pathAsString.hashCode()
    }

    companion object {
        fun from(pathAsString: String): Path {
            return PathImpl(pathAsString)
        }
    }
}
